### PR TITLE
common: remove support for SyCL older than 2020

### PR DESCRIPTION
**Pull request by spalicki**
Removing SYCL 1.2.1 specific code.

### CODE REVIEW

**Review by TaoLv — APPROVED** [review]

File: src/graph/utils/sycl_check.hpp [hunk 1]
```diff
@@ -26,18 +26,8 @@
...
-#define DNNL_USE_SYCL121_API 0
-#endif
-#elif defined(__LIBSYCL_MAJOR_VERSION)
-#if (__LIBSYCL_MAJOR_VERSION < 6)
-#define DNNL_USE_SYCL121_API 1
-#else
-#define DNNL_USE_SYCL121_API 0
-#endif
+#if defined(__INTEL_LLVM_COMPILER) && __INTEL_LLVM_COMPILER >= 20230000
+#elif defined(__LIBSYCL_MAJOR_VERSION) && __LIBSYCL_MAJOR_VERSION >= 6
```
Comment (mgouicem) [inline src/graph/utils/sycl_check.hpp:30]: (random) there might be other spots to edit:
- check SYCL_COMPILER_VERSION macro
- there might be some spots in cuda/hip backend as well where workarounds were implemented (tagging @t4c1 @sgeor255).

**Review by rjoursler — APPROVED** [review]

**Comment by vpirogov** [comment]
make test

File: src/xpu/sycl/compat.hpp [hunk 2]
```diff
@@ -58,20 +58,6 @@ inline void host_task(H &cgh, F &&f) {
...
-        = ::sycl::info::device::ext_intel_gpu_slices;
-constexpr auto ext_intel_gpu_subslices_per_slice
-        = ::sycl::info::device::ext_intel_gpu_subslices_per_slice;
-
-const auto cpu_selector_v = ::sycl::cpu_selector {};
-const auto gpu_selector_v = ::sycl::gpu_selector {};
-#else
-
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (densamoilov) [inline src/xpu/sycl/compat.hpp:62]: Can we stop suppressing the warning now?

**Comment by spalicki** [comment]
@densamoilov @mgouicem I also wanted to remove CL/sycl.hpp header include since it is there just for compatibility with older SyCL versions. In many places we have:
```
#if __has_include(<sycl/sycl.hpp>)
#include <sycl/sycl.hpp>
#elif __has_include(<CL/sycl.hpp>)
#include <CL/sycl.hpp>
#else
#error "Unsupported compiler"
#endif
```
I wanted to simplify it to:
```
#if __has_include(<sycl/sycl.hpp>)
#include <sycl/sycl.hpp>
#else
#error "Unsupported compiler"
#endif
```

**Comment by densamoilov** [comment]
>  I also wanted to remove CL/sycl.hpp header include since it is there just for compatibility with older SyCL versions. 

@spalicki, I think it should be fine.

**Comment by spalicki** [comment]
make test

**Comment by spalicki** [comment]
make test

**Comment by spalicki** [comment]
make test

File: src/graph/utils/sycl_check.hpp [hunk 3]
```diff
@@ -20,24 +20,12 @@
...
-#else
-#define DNNL_USE_SYCL121_API 0
-#endif
-#elif defined(__LIBSYCL_MAJOR_VERSION)
-#if (__LIBSYCL_MAJOR_VERSION < 6)
-#define DNNL_USE_SYCL121_API 1
-#else
-#define DNNL_USE_SYCL121_API 0
-#endif
+#if defined(__INTEL_LLVM_COMPILER) && __INTEL_LLVM_COMPILER >= 20230000
```
Comment (densamoilov) [inline src/graph/utils/sycl_check.hpp:27]: Why do we keep this code?

Comment (spalicki) [reply src/graph/utils/sycl_check.hpp:27]: Theoretically now that I removed the `CL/sycl.hpp` headers `#if __has_include(<sycl/sycl.hpp>)` implied high enough SyCL version, so this check can be removed as redundant.

**Comment by spalicki** [comment]
make test

File: src/xpu/sycl/compat.hpp [hunk 4]
```diff
@@ -58,36 +58,17 @@ inline void host_task(H &cgh, F &&f) {
...
-        = ::sycl::info::device::ext_intel_gpu_slices;
-constexpr auto ext_intel_gpu_subslices_per_slice
-        = ::sycl::info::device::ext_intel_gpu_subslices_per_slice;
-
-const auto cpu_selector_v = ::sycl::cpu_selector {};
-const auto gpu_selector_v = ::sycl::gpu_selector {};
-#else
-
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (mgouicem) [inline src/xpu/sycl/compat.hpp:76]: this diagnostic ignore is in else path, did you remove it on purpose?

**Review by mgouicem — APPROVED** [review]

File: src/xpu/sycl/compat.hpp (hunk 4 above)
Comment (spalicki) [reply src/xpu/sycl/compat.hpp:76]: Yes, it seems to be no longer needed. Everything compiles without warnings without it.

**Review by hidefromkgb — APPROVED** [review]

**Pull request merged**